### ISSUE TITLE
9.2.0 broke `sql-sync`

### ISSUE THREAD

**Issue by grasmash** [drush-ops/drush#3417]
My test suites for acquia/blt began failing immediately after drush/drush 9.2.0 was released. 

Executing:
```
./vendor/bin/drush sql-sync @default.clone @self --structure-tables-key=lightweight --create-db --no-interaction --ansi
```

Outputs:
```
You will destroy data in drupal and replace with data from drupal3.
 [notice] Starting to create database on target.
Creating database drupal. Any existing database will be dropped!
 // Do you really want to continue?: yes.                                       
 [notice] Starting to dump database on source.
 [success] Database dump saved to 1.gz
 [notice] Starting to import dump file onto target database.
 [error]  File(s) not found: 1.gz 
 [error]  Failed to import 1.gz into target. 
```

**Comment by grasmash** [comment]
I see that some related things changed in 9.1.0...9.2.2:
https://github.com/drush-ops/drush/compare/9.1.0...9.2.0#diff-8a108c18009fb8a2326f1fe42f0c3ebfR41

**Comment by weitzman** [comment]
Compare link is broken. Perhaps add -vvv for more debug info. A bit hard to know whats wrong from the provided log. Also helpful to share output of drush sa `@default.clone` and `drush status`

**Comment by rodrigoaguilera** [comment]
I have been trying to debug this for the last hours I have nothing except that "1.gz" should be DATABASENAME_DATE.gz

**Comment by greg-1-anderson** [comment]
sql-sync should be getting the filename from the result of the sql-dump command.

**Comment by rodrigoaguilera** [comment]
For more context the sites that I am trying to sync both have local aliases

**Comment by weitzman** [comment]
That filename was a good clue!. Fix in #3418.

FYI the behavior of --no-interaction was fixed in this release as well so please be on the lookout for problems there.

Closing this in favor of #3418
